apollo-router/tests/integration/entity_cache.rs

```diff
@@ -0,0 +1,985 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use apollo_compiler::Schema;
+use apollo_router::_private::MockStore;
+use apollo_router::_private::RedisCacheStorage;
+use apollo_router::_private::entity_cache::CONTEXT_CACHE_KEYS;
+use apollo_router::_private::entity_cache::CacheKeyContext;
+use apollo_router::_private::entity_cache::CacheKeysContext;
+use apollo_router::_private::entity_cache::EntityCache;
+use apollo_router::_private::entity_cache::Subgraph;
+use apollo_router::Context;
+use apollo_router::MockedSubgraphs;
+use apollo_router::TestHarness;
+use apollo_router::plugin::test::MockSubgraph;
+use apollo_router::plugin::test::MockSubgraphService;
+use apollo_router::services::subgraph;
+use apollo_router::services::supergraph;
+use http::HeaderValue;
+use http::header::CACHE_CONTROL;
+use tower::Service;
+use tower::ServiceExt;
+
+const SCHEMA: &str = include_str!("../../src/testdata/orga_supergraph.graphql");
+const SCHEMA_REQUIRES: &str = include_str!("../../src/testdata/supergraph.graphql");
+
+#[tokio::test]
+async fn insert() {
+    let valid_schema = Arc::new(Schema::parse_and_validate(SCHEMA, "test.graphql").unwrap());
+    let query = "query { currentUser { activeOrganization { id creatorUser { __typename id } } } }";
+
+    let redis_cache = RedisCacheStorage::from_mocks(Arc::new(MockStore::default()))
+        .await
+        .unwrap();
+    let map = [
+        (
+            "user".to_string(),
+            Subgraph {
+                redis: None,
+                private_id: Some("sub".to_string()),
+                enabled: true,
+                ttl: None,
+                ..Default::default()
+            },
+        ),
+        (
+            "orga".to_string(),
+            Subgraph {
+                redis: None,
+                private_id: Some("sub".to_string()),
+                enabled: true,
+                ttl: None,
+                ..Default::default()
+            },
+        ),
+    ]
+    .into_iter()
+    .collect();
+    let entity_cache = EntityCache::with_mocks(redis_cache.clone(), map, valid_schema.clone())
+        .await
+        .unwrap();
+
+    let mocks = serde_json::json!({
+        "user": {
+            "headers": {CACHE_CONTROL.as_str(): "public"},
+            "query": {
+                "currentUser": {
+                    "activeOrganization": {
+                        "__typename": "Organization",
+                        "id": "1",
+                    }
+                }
+            }
+        },
+        "orga": {
+            "headers": {CACHE_CONTROL.as_str(): "public"},
+            "entities": [
+                {
+                    "__typename": "Organization",
+                    "id": "1",
+                    "creatorUser": {
+                        "__typename": "User",
+                        "id": 2
+                    }
+                }
+            ]
+        },
+    });
```

Comment:
Parsing a YAML string could be nicer (less punctuation heavy), I stuck with `json!` macro here because that’s what we do everywhere for router config in `TestHarness` tests